scipy/sparse/coo.py

```diff
@@ -190,6 +190,31 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
 
         self._check()
 
+    def reshape(self, shape, order='C', copy=False):
```

Comment:
You should also allow the *args shape specification

Reply:
I don't feel strongly either way, but I am disinclined to add it until Numpy does it first to either the `reshape` function or `reshape` method on `ndarray`.

Reply:
But it has that functionality... X.reshape(5,6)

Reply:
Huh, you are correct. That's apparently undocumented. I am not even sure what the exact behavior is.

Reply:
It appears that `ndarray.reshape` first checks if there is more than one positional argument; if there is, it tries to cast all positional arguments to ints and joins them into a tuple. If there is only one argument, it tries to iterate over it and cast all contents to ints and joins them into a tuple. If it succeeds in iterating, but fails at casting it raises the casting error. If iterating fails, it then tries to cast sole argument to an int and puts it in a tuple by itself.

The `shape` parameter is a positional-only argument. The `order` parameter is a named-only argument. This suggests the true signature is `reshape(shape*, order='C')`

Is there a function in Numpy somewhere that does this shape args unification so I can be sure I am consistent?